core/src/main/java/org/elasticsearch/action/ingest/IngestActionFilter.java

```diff
@@ -91,8 +92,8 @@ void processIndexRequest(Task task, String action, ActionListener listener, Acti
             chain.proceed(task, action, indexRequest, listener);
             return;
         }
-        executionService.execute(indexRequest, pipelineId, t -> {
-            logger.error("failed to execute pipeline [{}]", t, pipelineId);
+        executionService.execute(indexRequest, t -> {
+            logger.error("failed to execute pipeline [{}]", t, indexRequest.pipeline());
             listener.onFailure(t);
         }, success -> {
             indexRequest.putHeader(PIPELINE_ALREADY_PROCESSED, true);
```

Comment:
this would have a downside: on a node client, we would modify the original request by removing the pipeline id which would be weird to users, unless we create a copy of the index request. I haven't decided yet which approach is cleaner... maybe the header is not that bad after all :)